packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -684,6 +691,102 @@ describe('SignatureController', () => {
     });
   });
 
+  describe('trySetDeferredSignSuccess', () => {
+    it('sets a message status as signed in a message manager', () => {
+      signatureController.setDeferredSignSuccess(
+        messageParamsMock.metamaskId,
+        messageParamsMock.data,
+      );
+
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledTimes(
+        1,
+      );
+      expect(messageManagerMock.setMessageStatusSigned).toHaveBeenCalledWith(
+        messageIdMock,
+        messageParamsWithoutIdMock.data,
+      );
+
+      expect(
+        personalMessageManagerMock.setMessageStatusSigned,
+      ).not.toHaveBeenCalled();
+      expect(
+        typedMessageManagerMock.setMessageStatusSigned,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('should return false when an error occurs', () => {
```

Comment:
As with the error method, do we also need tests to verify it moves onto the next message manager if the first throws, and also once that verifies it throws if all the message managers throw?

Reply:
Imo we're good, and it would be almost to test that the for loop works as intended, it will automatically continue to the next iteration and check all the managers until it finds a successful match or completes the loop.

Reply:
I just mean that we while we know the logic is the same underneath, we have to write these unit tests as a black box and currently there's no test checking that this method will throw an exception or even support more than just the initial message manager.